kubernetes/kubernetes.go

```diff
@@ -236,8 +241,8 @@ func newKubeAgent(config KubeAgentConfig) KubeAgentConfig {
 	return config
 }
 
-func (ka KubeAgentConfig) collectMetrics(
-	config KubeAgentConfig, clientset kubernetes.Interface, nodeSource NodeSource) (rerr error) {
+func (ka KubeAgentConfig) collectMetrics(ctx context.Context, config KubeAgentConfig,
```

Comment:
Looks like this method has developed a code smell (needs to be refactored to reduce complexity)

Reply:
Thanks for catching! I'll take a look and update.

Reply:
Hmm looking at those code smells Sonar threw, fixing them doesn't seem within the scope of this PR. Looks like they weren't caught with the initial writing of that function (or maybe if/when someone added to it?). To fix each would divert time and resources away from the 1.21 work. I agree with you - ideally we'd fix and not have large functions like this in the codebase. Given it's lower priority than getting the 1.21 work out the door, I'm going to punt on this for now. If there's time later, we can revisit.

Code smells:
![Screen Shot 2021-09-09 at 11 03 43 AM](https://user-images.githubusercontent.com/15006388/132739124-d639f710-aab1-469a-81fe-96e582283d42.png)
![Screen Shot 2021-09-09 at 11 03 52 AM](https://user-images.githubusercontent.com/15006388/132739130-6214542a-a642-4648-b954-a8ec96d00b8a.png)